packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts

```diff
@@ -235,7 +239,18 @@ class DataStorageAuthWithTriggerTestProject extends TestProjectBase {
     await this.checkLambdaResponse(node16Lambda[0], expectedResponse);
     await this.checkLambdaResponse(funcWithSsm[0], 'It is working');
     await this.checkLambdaResponse(funcWithAwsSdk[0], 'It is working');
-    await this.checkLambdaResponse(funcWithSchedule[0], 'It is working');
+
+    // test schedule function event trigger once
+    if (this.checkInvocationCount) {
```

Comment:
I'm assuming this is here to prevent duplicate checking, but I'm confused why it's necessary?

Reply:
This reduced a lot of flakiness that I was seeing by only testing for invocation count on the first deployment instead of also on subsequent sandbox updates. This will also help us reduce time on e2e tests as we don't have to wait 70 seconds for each `assertPostDeployment`.

Reply:
Sounds good. Could be helpful to leave that context in a comment on this code.

Reply:
Done in https://github.com/aws-amplify/amplify-backend/pull/1527/commits/ca2d0bd1eb466fc5752cb4b830375af273372dec